kombu/connection.py

```diff
@@ -279,8 +279,8 @@ def _debug(self, msg, *args, **kwargs):
 
     def connect(self):
         """Establish connection to server immediately."""
-        self._closed = False
-        return self.connection
+        conn_opts = self._extract_failover_opts()
+        return self._ensure_connection(**conn_opts)
```

Comment:
@matusvalo this now creates and returns a new connection, but doesn't set `self._connection`
```
conn = Connection(...)
conn.connect()
conn.connected -> False
```

Reply:
This broke our status check flow

Reply:
can yu check new commits on master?